src/GazeboThruster.cpp

```diff
@@ -140,6 +148,16 @@ void GazeboThruster::readInput(ThrustersMSG const& thrustersMSG)
     }
 }
 
+void GazeboThruster::readCompensatedMass(CompMassMSG const& compMassMSG)
+{
+    gazebo_underwater::Matrix6 matrix(compMassMSG->matrix());
+    gazebo::math::Vector3 cog(gazebo::msgs::ConvertIgn(compMassMSG->cog()));
+    if(!compensated_effort)
```

Comment:
Can't you compare the current mass with the former one at least ? It would be a lot more robust ...